spec/requests/api/v1/customers_spec.rb

```diff
@@ -131,13 +131,172 @@
+        groups_usage = charge_usage[:groups]
+
+        aggregate_failures do
+          expect(charge_usage[:units]).to eq('4.0')
+          expect(charge_usage[:amount_cents]).to eq(5000)
+          expect(groups_usage).to match_array(
+            [
+              { lago_id: aws.id, key: nil, value: 'aws', units: '3.0', amount_cents: 3000 },
+              { lago_id: google.id, key: nil, value: 'google', units: '1.0', amount_cents: 2000 },
+            ],
+          )
+        end
+      end
+    end
+
+    context 'with two dimensions group' do
+      let(:aws) { create(:group, billable_metric: metric, key: 'cloud', value: 'aws') }
+      let(:google) { create(:group, billable_metric: metric, key: 'cloud', value: 'google') }
+      let(:aws_usa) { create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: aws.id) }
+      let(:aws_france) { create(:group, billable_metric: metric, key: 'region', value: 'france', parent_group_id: aws.id) }
+      let(:google_usa) { create(:group, billable_metric: metric, key: 'region', value: 'usa', parent_group_id: google.id) }
+
+      let(:charge) do
+        create(
+          :standard_charge,
+          plan: subscription.plan,
+          billable_metric: metric,
+          properties: {},
+          group_properties: [
+            build(
+              :group_property,
+              group: aws_usa,
+              values: { amount: '10', amount_currency: 'EUR' },
+            ),
+            build(
+              :group_property,
+              group: aws_france,
+              values: { amount: '20', amount_currency: 'EUR' },
+            ),
+            build(
+              :group_property,
+              group: google_usa,
+              values: { amount: '30', amount_currency: 'EUR' },
+            ),
+          ],
+        )
+      end
+
+      before do
+        create_list(
+          :event,
+          2,
+          organization: organization,
+          customer: customer,
+          subscription: subscription,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'aws', region: 'usa' },
+        )
+
+        create(
+          :event,
+          organization: organization,
+          customer: customer,
+          subscription: subscription,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'aws', region: 'france' },
+        )
+
+        create(
+          :event,
+          organization: organization,
+          customer: customer,
+          subscription: subscription,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'google', region: 'usa' },
+        )
+      end
+
+      it 'returns the group usage for the customer' do
+        get_with_token(
+          organization,
+          "/api/v1/customers/#{customer.external_id}/current_usage?external_subscription_id=#{subscription.external_id}",
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)